proc_nutation.py

```diff
@@ -1,53 +1,84 @@
+from pylab import *
 from pyspecdata import *
 from scipy.optimize import minimize
-from proc_scripts import hermitian_function_test,zeroth_order_ph,postproc_dict,fl_mod,center_CPMG_echo
+from proc_scripts import hermitian_function_test,zeroth_order_ph,postproc_dict,fl_mod
 from sympy import symbols
 from numpy import *
 fl = fl_mod()
 t2 = symbols('t2')
 logger = init_logging("info")
-for searchstr,exp_type,nodename,postproc in [
-    ['200219_nutation_alex_probe','test_equip','nutation','spincore_nutation_v1']
+max_kHz = 200
+for searchstr,exp_type,nodename,postproc,freq_slice,p90_varied in [
+    ['201211_Ni_sol_probe_nutation_1','nutation','nutation',
+        'spincore_nutation_v1',(-5000,13000),True],
+    #['210302_Ni_cap_probe_nutation_1','nutation','nutation',
+    #    'spincore_nutation_v1',(-4e3,1.2e4),True]
     ]:
     s = find_file(searchstr,exp_type=exp_type,expno=nodename,postproc=postproc,
-            lookup=postproc_dict) 
- 
+            lookup=postproc_dict)#,fl=fl) 
+    if not p90_varied:
+        plen = s.get_prop('acq_params')['p90_us']
+        plen *= 10**-6
     # {{{ do the rough centering before anything else!
     # in particular -- if you don't do this before convolution, the
     # convolution doesn't work properly!
-    s.ft(['ph2','ph1'])
-    # }}}
-    
+    s.ift('t2')
+    s.setaxis('t2', lambda x: x-abs(s['ph1',1]['ph2',-2]).mean_all_but('t2').argmax('t2').item())
+    #}}}
+    fl.next('t domain centered')
+    fl.image(s)
+    fl.next('freq domain centered')
+    s.ft('t2')
+    fl.image(s)
+    s.ift('t2')
     # {{{ centering of data using hermitian function test
-    best_shift = hermitian_function_test(s['ph2',0]['ph1',1])
-    logger.info(strm("best shift is",best_shift))
-    s.setaxis('t2', lambda x: x-best_shift).register_axis({'t2':0})
+    if p90_varied:
+        best_shift = hermitian_function_test(s['ph2',0]['ph1',1])
+        logger.info(strm("best shift is",best_shift))
+        s.setaxis('t2', lambda x: x-best_shift).register_axis({'t2':0})
+        fl.next('with time shift')
+        fl.image(s)
+        fl.next('freq domain after time correction')
+        s.ft('t2')
+        fl.image(s)
```

Comment:
I don't understand why you are only running the hermitian function test if you have a variable p90?

Reply:
which one?

Reply:
not resolved, but will start a new comment

Reply:
when I attempt to apply the hermitian function test to the amp varied data (I also reworded your comment to correct that it is the amp varied where hermitian is not applied) I get this error
`Traceback (most recent call last):
  File "proc_nutation.py", line 26, in <module>
    best_shift = hermitian_function_test(s['ph1',1]['ph2',0])
  File "C:\Users\Alex Guinness\gitstuff\proc_scripts\proc_scripts\phasing.py", line 223, in hermitian_fu
nction_test
    s.ift('t2')
  File "c:\users\alex guinness\gitstuff\pyspecdata\pyspecdata\fourier\ift.py", line 179, in ift
    self._ft_shift(thisaxis,p2_pre)
  File "c:\users\alex guinness\gitstuff\pyspecdata\pyspecdata\fourier\ft_shift.py", line 77, in _ft_shif
t
    newdata = np.empty_like(self.data)
  File "<__array_function__ internals>", line 5, in empty_like
MemoryError: Unable to allocate 14.6 GiB for an array with shape (199, 4096, 1200) and data type complex
128` 
I don't understand why unless there is just a boat load of more data in the amp varied compared to the p90 varied? I have left my trouble shooting code in place so you can get an idea of how I went about it and maybe I am missing something or going about it the wrong way?